src/lib.rs

```diff
@@ -60,29 +74,49 @@ impl ScuttleButt {
     pub fn self_node_state(&mut self) -> &mut NodeState {
         self.cluster_state_map.node_state_mut(&self.self_node_id)
     }
+
+    /// Retrieve a list of all living nodes.
+    pub fn living_nodes(&self) -> impl Iterator<Item = &str> {
+        self.cluster_state_map.living_nodes()
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
+    use std::collections::HashMap;
+    use std::fmt::Debug;
+    use std::hash::Hash;
+
     fn run_scuttlebutt_handshake(initiating_node: &mut ScuttleButt, peer_node: &mut ScuttleButt) {
         let syn_message = initiating_node.create_syn_message();
         let syn_ack_message = peer_node.process_message(syn_message).unwrap();
         let ack_message = initiating_node.process_message(syn_ack_message).unwrap();
         assert!(peer_node.process_message(ack_message).is_none());
     }
 
-    fn assert_nodes_sync(nodes: &[&ScuttleButt]) -> bool {
-        if nodes.len() <= 1 {
-            return true;
+    fn assert_map_eq<K, V>(lhs: &HashMap<K, V>, rhs: &HashMap<K, V>)
+    where
+        K: Eq + Hash,
+        V: PartialEq + Debug,
+    {
+        assert_eq!(lhs.len(), rhs.len());
+        for (key, value) in lhs {
+            assert_eq!(rhs.get(key), Some(value));
         }
-        let first_node_state = serde_json::to_value(&nodes[0].cluster_state_map).unwrap();
+    }
+
+    fn assert_nodes_sync(nodes: &[&ScuttleButt]) {
+        let first_node_states = &nodes[0].cluster_state_map.node_states;
         for other_node in nodes.iter().skip(1) {
-            let node_state = serde_json::to_value(&other_node.cluster_state_map).unwrap();
-            assert_json_diff::assert_json_eq!(&first_node_state, &node_state);
+            let node_states = &other_node.cluster_state_map.node_states;
+
+            assert_eq!(first_node_states.len(), node_states.len());
+            for (key, value) in first_node_states {
+                assert_map_eq(&value.key_values, &node_states.get(key).unwrap().key_values);
+            }
```

Comment:
This should be a pretty simple replacement for the json conversion without requiring serde. Only tests the actual K/V pairs instead of the entire node state, but I think that's the most important part here and the verbosity of testing every single field isn't beneficial.

Reply:
yes